packages/profile-sync-controller/src/controllers/user-storage/account-syncing/controller-integration.ts

```diff
@@ -25,26 +23,53 @@ export async function saveInternalAccountToUserStorage(
   internalAccount: InternalAccount,
   options: AccountSyncingOptions,
 ): Promise<void> {
-  const { getUserStorageControllerInstance } = options;
+  const { getUserStorageControllerInstance, getMessenger } = options;
 
   if (
     !canPerformAccountSyncing(options) ||
-    !isEvmAccountType(internalAccount.type) ||
-    !(await isInternalAccountFromPrimarySRPHdKeyring(internalAccount, options))
+    internalAccount.metadata.keyring.type !== String(KeyringTypes.hd) // sync only EVM accounts until we support multichain accounts
   ) {
     return;
   }
 
+  // Refresh the internal accounts list so that it populates entropySourceId and derivationPath for all accounts
+  await getMessenger().call('AccountsController:updateAccounts');
+  // Find back our account from this refreshed list
+  const internalAccountsList = getMessenger().call(
+    'AccountsController:listAccounts',
+  );
+
+  const internalAccountFromList = internalAccountsList.find(
+    (account) => account.address === internalAccount.address,
+  );
+
+  // This should never happen because users don't have the ability to remove an accounts
+  // but if it does, we throw an error
+  // istanbul ignore next
+  if (!internalAccountFromList) {
+    throw new Error(
+      `UserStorageController - failed to find internal account in the list - ${internalAccount.address}`,
+    );
+  }
+
+  const entropySourceId =
+    String(
+      JSON.stringify(internalAccountFromList.options.entropySource).replace(
+        /"/gu,
+        '',
+      ),
+    ) || undefined;
```

Comment:
Why not something like this?
```suggestion
  const entropySourceId = internalAccountFromList.options.entropySource?.toString();
```

Reply:
Am I the only one bothered that `InternalAccount` is not actually defined anywhere?
It's defined as the return type of a method that defines its return type as `InternalAccount` :|
That seems to be the reason for the wrong type for `options`, at first glance.

aaanyway, casting to `string | undefined` should be enough here